src/main/java/stringcalculator/StringCalculator.java

```diff
@@ -0,0 +1,63 @@
+package stringcalculator;
+
+import org.apache.logging.log4j.util.Strings;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class StringCalculator {
+    private final Pattern customSeparatorPattern = Pattern.compile("//(.+?)\\n(.*)");
+    private final List<String> defaultSeparators = List.of(":", ",");
+
+    public StringCalculator() {
+    }
+
+    public Integer calculate(String input) {
+        if (Strings.isEmpty(input)) {
+            return 0;
+        }
+
+        List<String> separators = new ArrayList<>(this.defaultSeparators);
+        SeparatorParser separatorParser = new SeparatorParser(this.customSeparatorPattern);
+        CustomSeparator customSeparator = separatorParser.parseSeparator(input);
+
+        if (customSeparator.isParsedStatus()) {
```

Comment:
커스텀 구분자가 파싱된 상태인지 검사하지않고
SeparatorParser 에게 input내부에 커스텀 구분자가 존재하는지 물어보고나서 `parseSeparator` 를 호출하는것은 어떻게 생각하시나요?

일단 파싱하고나서 `파싱했어?` 라고 객체에게 물어보는것 보다
`커스텀 구분자 있어?` 라고 물어본 뒤 진행하는게 더 깔끔하다고 생각해서요! 

```suggestion
        SeparatorParser separatorParser = new SeparatorParser(this.customSeparatorPattern);
        if (separatorParser.hasCustomSeparator(input)) {
            CustomSeparator customSeparator = separatorParser.parseSeparator(input);
            // .. 생략
        }
```